### PR TITLE
document building with ninja + non interactive options

### PR DESCRIPTION
**Pull request by tudorbarascu**

### CODE REVIEW

File: INSTALL [hunk 1]
```diff
@@ -262,27 +265,55 @@ interactive dialog.
...
   export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:${HOME}/apps/lib/
 
+Optionally, if you already know what aspects you want in your custom build
+then you can skip the interactive ccmake .. part by using the cmake -D
+option for each aspect, e.g.:
+
+  cmake -D CMAKE_BUILD_TYPE=Debug -D CMAKE_INSTALL_PREFIX=${HOME}/apps ..
+
+Also, if you want to speed your build times, you can easily do it with Ninja
+as it's very similar to cmake.
```
Comment (m-kuhn) [inline INSTALL]: `ninja` is an alternative to `make`

**Review by m-kuhn — CHANGES_REQUESTED** [review]
Thanks, a very good enhancement!
Just some minor comments from my side

File: INSTALL [hunk 2]
```diff
@@ -262,27 +265,55 @@ interactive dialog.
...
+
+For example, to configure your build you can do either one of:
+
+  ccmake -GNinja ..
+
+  cmake -G Ninja -D CMAKE_BUILD_TYPE=Debug -D CMAKE_INSTALL_PREFIX=${HOME}/apps ..
+
+Build and install with ninja:
+
+  ninja (also supports the above described -jX option)
```
Comment (m-kuhn) [inline INSTALL]: ninja uses all cores by default, I think the comment should be something like "uses all cores by default".

**Comment by tudorbarascu** [comment]
Thanks for the comments! changes implemented.

File: INSTALL [hunk 3]
```diff
@@ -262,27 +265,55 @@ interactive dialog.
 
 Now on with the build:
 
-  make
-  make install
+  make -jX
+
+where X is the number of available cores. Without the -j option
+it may take a little while to build depending on your platform.
```
Comment (m-kuhn) [inline INSTALL]: Even with the -j option it may take a little while ;)

File: INSTALL [hunk 4]
```diff
@@ -262,27 +265,55 @@ interactive dialog.
...
+  cmake -D CMAKE_BUILD_TYPE=Debug -D CMAKE_INSTALL_PREFIX=${HOME}/apps ..
+
+Also, if you want to speed your build times, you can easily do it with ninja,
+an alternative to make with similar build options.
+
+For example, to configure your build you can do either one of:
+
+  ccmake -GNinja ..
+
+  cmake -G Ninja -D CMAKE_BUILD_TYPE=Debug -D CMAKE_INSTALL_PREFIX=${HOME}/apps ..
```
Comment (m-kuhn) [inline INSTALL]: Does `-G Ninja` work as well or is this a typo? I always use the version from two lines above.

Comment (tudorbarascu) [inline INSTALL]: it works as well, ``ccmake -G Ninja ..`` also works as well. Should I make it consistent? (I'm leaning towards ``-G Ninja``...

Comment (m-kuhn) [inline INSTALL]: I think consistency would be good, no preference from my side, go for the one you prefer

**Comment by m-kuhn** [comment]
Nice,

one last thing I spotted, not sure if it makes a difference, but I think using `--encoding=utf-8` won't hurt

**Comment by m-kuhn** [comment]
Thanks @tudorbarascu

**Pull request merged**